Applications/Utils/SimpleMeshCreation/createMeshElemPropertiesFromASCRaster.cpp

```diff
@@ -35,212 +36,158 @@
-    BaseLib::quicksort<double>(src_properties, 0, size, src_perm);
-
-    // compress the property data structure
-    const std::size_t mat_map_size(src_properties.size());
-    std::vector<std::size_t> mat_map(mat_map_size);
-    mat_map[0] = 0;
-    std::size_t n_mat(1);
-    for (std::size_t k(1); k<mat_map_size; ++k) {
-        if (std::fabs(src_properties[k] - src_properties[k-1]) > std::numeric_limits<double>::epsilon()) {
-            mat_map[k] = mat_map[k - 1] + 1;
-            n_mat++;
-        } else
-            mat_map[k] = mat_map[k - 1];
-    }
-    std::vector<double> compressed_src_properties(n_mat);
-    compressed_src_properties[0] = src_properties[0];
-    for (std::size_t k(1), id(1); k<mat_map_size; ++k) {
-        if (std::fabs(src_properties[k] - src_properties[k-1]) > std::numeric_limits<double>::epsilon()) {
-            compressed_src_properties[id] = src_properties[k];
-            id++;
-        }
-    }
-    compressed_src_properties[n_mat - 1] = src_properties[mat_map_size - 1];
+    // do the interpolation
+    MeshLib::Mesh2MeshPropertyInterpolation mesh_interpolation(
+        src_mesh.get(), property_arg.getValue());
+    mesh_interpolation.setPropertiesForMesh(dest_mesh.get());
 
-    // reset materials in source mesh
-    const std::size_t n_mesh_elements(src_mesh->getNumberOfElements());
-    auto materialIds = src_mesh->getProperties().getPropertyVector<int>("MaterialIDs");
-    if (!materialIds)
+    double scale(1.0);
+    if (unit_arg.getValue() == "m/s")
     {
-        materialIds = src_mesh->getProperties().createNewPropertyVector<int>
-            ("MaterialIDs", MeshLib::MeshItemType::Cell, 1);
-        materialIds->resize(n_mesh_elements, 0);    // default material id
+        scale = 1.0;
     }
-    for (std::size_t k(0); k<n_mesh_elements; k++)
-        (*materialIds)[src_mesh->getElement(src_perm[k])->getID()] = mat_map[k];
-
-    // do the interpolation
-    MeshLib::Mesh2MeshPropertyInterpolation mesh_interpolation(src_mesh.get(),
-                                                               &compressed_src_properties);
-    std::vector<double> dest_properties(dest_mesh->getNumberOfElements());
-    mesh_interpolation.setPropertiesForMesh(dest_mesh.get(),
-                                            dest_properties);
-
-    const std::size_t n_dest_mesh_elements(dest_mesh->getNumberOfElements());
-
-    { // write property file
-        std::string property_fname(mapping_arg.getValue());
-        std::ofstream property_out(property_fname.c_str());
-        if (!property_out)
-        {
-            ERR("Could not open file %s for writing the mapping.", property_fname.c_str());
-            return EXIT_FAILURE;
-        }
-
-        for (std::size_t k(0); k < n_dest_mesh_elements; k++)
-            property_out << k << " " << dest_properties[k] << "\n";
-        property_out.close();
+    else if (unit_arg.getValue() == "mm/a")
+    {
+        scale = 1e-3 / (365.25 * 86400);
```

Comment:
I think the scaling from non-SI to SI units should be either done by a user-supplied scaling parameter or in some other tool, e.g., Paraview.